File: Editor/ControlTypeAttribute.vb
<AttributeUsage(AttributeTargets.Class)> _
Public Class ControlTypeAttribute
    Inherits System.Attribute

    Private m_controlType As String

    Sub New(ByVal controlType As String)
        m_controlType = controlType
    End Sub

    Public ReadOnly Property ControlType() As String
        Get
            Return m_controlType
        End Get
    End Property
End Class

File: legacy/EditorControls/ControlTypeAttribute.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace TextAdventures.Quest.EditorControls
{
    [AttributeUsage(AttributeTargets.Class)]
    class ControlTypeAttribute : Attribute
    {
        public ControlTypeAttribute(string controlType)
        {
            ControlType = controlType;
        }

        public string ControlType { get; private set; }
    }
}
